include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -58,64 +110,77 @@ namespace seqan3
+ *          pos_it)
  * \brief   Erase a seqan3::gap from an aligned sequence.
  * \relates seqan3::aligned_sequence_concept
  *
- * \tparam        seq_type Type of the range to modify; must model
- *                         seqan3::aligned_sequence_concept.
- * \param[in,out] seq      The aligned sequence to modify.
- * \param[in]     pos_it   The iterator pointing to the position where to erase a gap.
+ * \tparam        aligned_seq_t   Type of the range to modify; must model seqan3::aligned_sequence_concept.
+ * \param[in,out] aligned_seq     The aligned sequence to modify.
+ * \param[in]     pos_it          The iterator pointing to the position where to erase a gap.
  *
  * \throws seqan3::gap_erase_failure if there is no seqan3::gap at \p pos_it.
  *
  * \details
  * \attention This is a concept requirement, not an actual function (however types
  *            modelling this concept will provide an implementation).
  */
-/*!\fn      inline seq_type::iterator erase_gap(seq_type & seq, typename seq_type::const_iterator first, typename seq_type::const_iterator last)
+/*!\fn      inline aligned_seq_t::iterator erase_gap(aligned_seq_t & aligned_seq, typename aligned_seq_t::const_iterator
+ *          first, typename aligned_seq_t::const_iterator last)
  * \brief   Erase multiple seqan3::gap from an aligned sequence.
  * \relates seqan3::aligned_sequence_concept
  *
- * \tparam        seq_type Type of the range to modify; must model
- *                         seqan3::aligned_sequence_concept.
- * \param[in,out] seq      The aligned sequence to modify.
- * \param[in]     first    The iterator pointing to the position where to start erasing gaps.
- * \param[in]     last     The iterator pointing to the position where to stop erasing gaps.
+ * \tparam        aligned_seq_t   Type of the range to modify; must model seqan3::aligned_sequence_concept.
+ * \param[in,out] aligned_seq     The aligned sequence to modify.
+ * \param[in]     first           The iterator pointing to the position where to start erasing gaps.
+ * \param[in]     last            The iterator pointing to the position where to stop erasing gaps.
  *
  * \throws seqan3::gap_erase_failure if one of the characters in [\p first, \p last) no seqan3::gap.
  *
  * \details
  * \attention This is a concept requirement, not an actual function (however types
  *            modelling this concept will provide an implementation).
  */
+/*!\fn      void assign_unaligned_sequence(aligned_seq_t & aligned_seq, unaligned_sequence_type && unaligned_seq)
+ *!\brief Assign an ungapped sequence to a gapped sequence.
+ * \relates seqan3::aligned_sequence_concept
+ *
+ * \tparam        aligned_seq_t     Type of the container to reassign; must model seqan3::sequence_container_concept;
+ *                                  the value type must be a seqan3::gapped alphabet.
+ * \tparam        unaligned_seq_t   Type of the container to assign from; must model std::ranges::InputRange;
```

Comment:
```suggestion
 * \tparam        unaligned_seq_t   Type of the container to assign from; must model std::ranges::ForwardRange;
```
?

Reply:
I think this statement is to generic. we need to formulate the requirements more clearly, that the type is expected to be the same as the aligned_seq_t without the gap information, I think. At least this is what the concept checks.